### PR TITLE
Project structure

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -175,16 +190,23 @@ $(HOME)/.transifexrc:
...
 .PHONY: transifex-init
-transifex-init: $(TX_DEPENDENCIES) c2cgeoportal/locale/c2cgeoportal.pot
+transifex-init: $(TX_DEPENDENCIES) \
+		geoportal/c2cgeoportal_geoportal/locale/c2cgeoportal_geoportal.pot \
+		admin/c2cgeoportal_admin/locale/c2cgeoportal_admin.pot
 	$(PRERULE_CMD)
-	tx push --source --force --no-interactive
-	tx push --translations --force --no-interactive
+	tx push --source --resource=geomapfish.c2cgeoportal_geoportal-$(TX_VERSION)
+	tx push --source --resource=geomapfish.c2cgeoportal_admin-$(TX_VERSION)
```
Comment (ger-benjamin) [inline Makefile]: No `--force --no-interactive` ?

File: commons/setup.py [hunk 2]
```diff
@@ -1,19 +1,57 @@
...
+    classifiers=[
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3",
+        "Framework :: Pyramid",
+        "Topic :: Internet :: WWW/HTTP",
+        "Topic :: Internet :: WWW/HTTP :: WSGI :: Application",
+    ],
     author="Camptocamp",
     author_email="info@camptocamp.com",
+    url="http://www.camptocamp.com/geospatial-solutions",
```
Comment (ger-benjamin) [inline commons/setup.py]: https://www.camptocamp.com/solutions/geospatial/ is better no ?

File: doc/integrator/make.rst [hunk 3]
```diff
@@ -76,21 +76,19 @@ Makefile config variables
...
-* ``POST_RULES``: Postdefine some build rules, default is empty.
-* ``PRE_RULES``: Predefine some build rules, default is empty.
+* ``NGEO_INTERFACES``: List of ngeo interfaces, default is ``mobile desktop``.
 * ``PRINT``: Mapfish print is enable, defaults to ``TRUE``.
 * ``MAPSERVER``: Mapserver is enable, defaults to ``TRUE``.
 * ``TILECLOUD_CHAIN``: ``TRUE`` to indicate that we use TileCloud-chain, defaults to ``TRUE``.
 * ``VISIBLE_ENTRY_POINT``: The entry point (path) of the application, defaults to ``/``.
 * ``VISIBLE_WEB_HOST``: The hostname, required.
-* ``VISIBLE_WEB_PROTOCOL``: The used protocole, defaults to ``https``.
+* ``VISIBLE_WEB_PROTOCOL``: The used protocol, defaults to ``https``.
```
Comment (ger-benjamin) [inline doc/integrator/make.rst]: s/defaults/default

File: docker/build/requirements.txt [hunk 4]
```diff
@@ -4,48 +4,51 @@ Babel==2.5.1
 boto==2.48.0
 bottle==0.12.13
 c2c.cssmin==0.7.dev6  # MakeFile
-c2c.template==1.5.1  # setup.py
+c2c.template==1.5.1  # geoportal
+http://github.com/camptocamp/c2cgeoform/archive/87a4191ef3330b76497bd009e9c0220cbc73c625.zip#egg=c2cgeoform  # commons
```
Comment (ger-benjamin) [inline docker/build/requirements.txt]: https ?

File: geoportal/c2cgeoportal_geoportal/lib/lingua_extractor.py [hunk 5]
```diff
@@ -207,17 +207,16 @@ def _collect_app_config(cls, config, filename):
...
+        models.init_dbsessions(settings)
+        from c2cgeoportal_geoportal.views.layers import Layers
         enums = []
         enum_layers = config["vars"].get("layers", {}).get("enum", {})
         for layername in list(enum_layers.keys()):
             layerinfos = enum_layers.get(layername, {})
             attributes = layerinfos.get("attributes", {})
             for fieldname in list(attributes.keys()):
-                values = cls._enumerate_attributes_values(DBSessions, Layers, layerinfos, fieldname)
+                values = cls._enumerate_attributes_values(models.DBSessions, Layers, layerinfos, fieldname)
```
Comment (ger-benjamin) [inline geoportal/c2cgeoportal_geoportal/lib/lingua_extractor.py]: Are you sure ?

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/geoportal/setup.py_tmpl [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+from setuptools import setup, find_packages
+
+setup(
+    name='{{package}}_geoportal',
+    version='1.0',
+    description='{{package}}, a c2cgeoportal project',
+    author='camptocamp',
+    author_email='info@camptocamp.com',
+    url='http://www.camptocamp.com/geospatial-solutions',
```
Comment (ger-benjamin) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/geoportal/setup.py_tmpl]: same as before

File: geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 7]
```diff
@@ -24,7 +24,7 @@ Version 2.3.0
 
 For non Docker upgrade have a look on the nondocker_CHANGELOG
 
-Remove the old configurations files:
-* rm -rf deploy
-* rm -rf apache
-* rm {package}/form.py
+1. The structure of the project change:
+   * The package name become `{{package}}_geoportal`.
+   * The package is in the `folder geoportal`.
```
Comment (ger-benjamin) [inline geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: `folder geoportal` -> folder `geoportal` (move the quote)

File: geoportal/c2cgeoportal_geoportal/scripts/c2cupgrade.py [hunk 8]
```diff
@@ -389,6 +389,15 @@ def files_to_move(self, pre=False):
                         error = True
                         managed = True
                         break
+                    if re.match(pattern + '$', dst):
+                        print(colorize(
+                            "The {} '{}' is present in the managed_files as '{}' but he will move.".format(
+                                type_, dst, pattern),
```
Comment (ger-benjamin) [inline geoportal/c2cgeoportal_geoportal/scripts/c2cupgrade.py]: risk of unicode issue ? (with dst ?)

File: geoportal/setup.py [hunk 9]
```diff
@@ -74,52 +77,52 @@
...
         "Programming Language :: Python",
-        "Framework :: Pylons",
+        "Programming Language :: Python :: 3",
+        "Framework :: Pyramid",
         "Topic :: Internet :: WWW/HTTP",
         "Topic :: Internet :: WWW/HTTP :: WSGI :: Application",
     ],
     author="Camptocamp",
     author_email="info@camptocamp.com",
     url="http://www.camptocamp.com/geospatial-solutions",
```
Comment (ger-benjamin) [inline geoportal/setup.py]: same as before

File: Makefile (hunk 1 above)
Comment (sbrunner) [reply Makefile]: oups missing :-)

File: commons/setup.py (hunk 2 above)
Comment (sbrunner) [reply commons/setup.py]: yes :-)

File: docker/build/requirements.txt (hunk 4 above)
Comment (sbrunner) [reply docker/build/requirements.txt]: +1

File: doc/integrator/make.rst (hunk 3 above)
Comment (sbrunner) [reply doc/integrator/make.rst]: thanks :-)

File: geoportal/c2cgeoportal_geoportal/lib/lingua_extractor.py (hunk 5 above)
Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/lib/lingua_extractor.py]: Yes :-)

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/geoportal/setup.py_tmpl (hunk 6 above)
Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/scaffolds/create/geoportal/setup.py_tmpl]: +1

File: geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 7 above)
Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: :-)

File: geoportal/c2cgeoportal_geoportal/scripts/c2cupgrade.py (hunk 8 above)
Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/scripts/c2cupgrade.py]: not in python3 :-)

File: geoportal/setup.py (hunk 9 above)
Comment (sbrunner) [reply geoportal/setup.py]: +1

File: geoportal/c2cgeoportal_geoportal/scripts/c2cupgrade.py (hunk 8 above)
Comment (ger-benjamin) [reply geoportal/c2cgeoportal_geoportal/scripts/c2cupgrade.py]: awww cool :tada:

**Pull request merged**